File: frontend/devpulse-ui/src/app/core/models/task-item.dto.ts
import { BaseDto } from './base.dto';

export interface TaskItemDto extends BaseDto {
  title: string;
  description: string;
  status: string;
  createdAt: Date;
  createdAtStr: string;
  isDeleted: boolean;
  isDeletedStr: string;

  userId: string;
  priority: string;
  dueDate: Date | null;
  dueDateStr: string;
}

export interface TaskItemWithUserDto extends TaskItemDto {
  userDisplayName: string;
}

File: backend/SharedLib/DTOs/Task/TaskItemDto.cs
namespace SharedLib.DTOs.Task
{
    public record TaskItemDto
    {
        public Guid Id { get; init; }
        public string Title { get; set; } = string.Empty;
        public string Description { get; set; } = string.Empty;
        public string Status { get; set; } = "Pending";
        public DateTime CreatedAt { get; set; }

        public string CreatedAtStr => CreatedAt.ToShortDateString();

        public bool IsDeleted { get; set; }
        public string IsDeletedStr => IsDeleted ? "Yes" : "No";

        public Guid UserId { get; set; }
        public string Priority { get; set; } = "Medium";
        public DateTime? DueDate { get; set; }
    }
}
